app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2145 @@
+            Authorization::skip(fn () => $dbForProject->increaseDocumentAttribute('topics', $topicId, 'total', 1));
+        } catch (DuplicateException) {
+            throw new Exception(Exception::SUBSCRIBER_ALREADY_EXISTS);
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($subscriber, Response::MODEL_SUBSCRIBER);
+    });
+
+App::get('/v1/messaging/topics/:topicId/subscribers')
+    ->desc('List topic\'s subscribers.')
+    ->groups(['api', 'messaging'])
+    ->label('scope', 'subscribers.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'listSubscribers')
+    ->label('sdk.description', '/docs/references/messaging/list-subscribers.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_SUBSCRIBER_LIST)
+    ->param('topicId', '', new UID(), 'Topic ID.')
+    ->param('queries', [], new Subscribers(), 'Array of query strings generated using the Query class provided by the SDK. [Learn more about queries](https://appwrite.io/docs/queries). Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' queries are allowed, each ' . APP_LIMIT_ARRAY_ELEMENT_SIZE . ' characters long. You may filter on the following attributes: ' . implode(', ', Providers::ALLOWED_ATTRIBUTES), true)
+    ->inject('dbForProject')
+    ->inject('response')
+    ->action(function (string $topicId, array $queries, Database $dbForProject, Response $response) {
+        $queries = Query::parseQueries($queries);
+
+        $topic = Authorization::skip(fn () => $dbForProject->getDocument('topics', $topicId));
+
+        if ($topic->isEmpty()) {
+            throw new Exception(Exception::TOPIC_NOT_FOUND);
+        }
+
+        \array_push($queries, Query::equal('topicInternalId', [$topic->getInternalId()]));
+
+        // Get cursor document if there was a cursor query
+        $cursor = Query::getByType($queries, [Query::TYPE_CURSORAFTER, Query::TYPE_CURSORBEFORE]);
+        $cursor = reset($cursor);
+
+        if ($cursor) {
+            $subscriberId = $cursor->getValue();
+            $cursorDocument = Authorization::skip(fn () => $dbForProject->getDocument('subscribers', $subscriberId));
+
+            if ($cursorDocument->isEmpty()) {
+                throw new Exception(Exception::GENERAL_CURSOR_NOT_FOUND, "Subscriber '{$subscriberId}' for the 'cursor' value not found.");
+            }
+
+            $cursor->setValue($cursorDocument);
+        }
+
+        $response
+            ->dynamic(new Document([
+                'subscribers' => $dbForProject->find('subscribers', $queries),
+                'total' => $dbForProject->count('subscribers', $queries, APP_LIMIT_COUNT),
+            ]), Response::MODEL_SUBSCRIBER_LIST);
+    });
+
+App::get('/v1/messaging/topics/:topicId/subscriber/:subscriberId')
```

Comment:
subscribers should be plural:

```suggestion
App::get('/v1/messaging/topics/:topicId/subscribers/:subscriberId')
```

Reply:
Done this